securedrop_client/gui/widgets.py

```diff
@@ -3681,7 +3681,7 @@ def __init__(self, source: Source, controller: Controller) -> None:
 class DeleteSourceAction(QAction):
     """Use this action to delete the source record."""
 
-    def __init__(self, source: Source, parent: SourceMenu, controller: Controller) -> None:
```

Comment:
I introduced this circular dependency when adding type hints. I thought narrower was better, and the circular dependency was hidden by the fact all `SourceMenu` and the actions are defined in the same file.